test/Test/MockApi.hs

```diff
@@ -156,39 +172,41 @@ instance (Algebra sig m, Has (Lift IO) sig m) => Algebra (MockApi :+: sig) (Mock
-matchExpectation a@(UploadArchive{}) (ApiExpectation _ b@(UploadArchive{}) resp) = resp <$ guard (a == b)
-matchExpectation a@(UploadContainerScan{}) (ApiExpectation _ b@(UploadContainerScan{}) resp) = resp <$ guard (a == b)
-matchExpectation a@(UploadContributors{}) (ApiExpectation _ b@(UploadContributors{}) resp) = resp <$ guard (a == b)
-matchExpectation a@(UploadLicenseScanResult{}) (ApiExpectation _ b@(UploadLicenseScanResult{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(AssertRevisionBinaries{}) (ApiExpectation _ _ b@(AssertRevisionBinaries{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(AssertUserDefinedBinaries{}) (ApiExpectation _ _ b@(AssertUserDefinedBinaries{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(FinalizeLicenseScan{}) (ApiExpectation _ _ b@(FinalizeLicenseScan{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetApiOpts) (ApiExpectation _ _ b@(GetApiOpts) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetAttribution{}) (ApiExpectation _ _ b@(GetAttribution{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetIssues{}) (ApiExpectation _ _ b@(GetIssues{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetLatestBuild{}) (ApiExpectation _ _ b@(GetLatestBuild{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetLatestScan{}) (ApiExpectation _ _ b@(GetLatestScan{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetOrganization{}) (ApiExpectation _ _ b@(GetOrganization{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetProject{}) (ApiExpectation _ _ b@(GetProject{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetScan{}) (ApiExpectation _ _ b@(GetScan{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(GetRevisionInfo{}) (ApiExpectation _ Yes b@(GetRevisionInfo{}) resp) = resp <$ guard (a == b)
+matchExpectation GetRevisionInfo{} (ApiExpectation _ No GetRevisionInfo{} resp) = Just resp
+matchExpectation a@(GetSignedLicenseScanUrl{}) (ApiExpectation _ _ b@(GetSignedLicenseScanUrl{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(ResolveProjectDependencies{}) (ApiExpectation _ _ b@(ResolveProjectDependencies{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(ResolveUserDefinedBinary{}) (ApiExpectation _ _ b@(ResolveUserDefinedBinary{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(UploadAnalysis{}) (ApiExpectation _ _ b@(UploadAnalysis{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(UploadArchive{}) (ApiExpectation _ _ b@(UploadArchive{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(UploadContainerScan{}) (ApiExpectation _ _ b@(UploadContainerScan{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(UploadContributors{}) (ApiExpectation _ _ b@(UploadContributors{}) resp) = resp <$ guard (a == b)
+matchExpectation a@(UploadLicenseScanResult{}) (ApiExpectation _ Yes b@(UploadLicenseScanResult{}) resp) = resp <$ guard (a == b)
+matchExpectation UploadLicenseScanResult{} (ApiExpectation _ No UploadLicenseScanResult{} resp) = Just resp
```

Comment:
I think I see how this `Yes`/`No` is useful here because we don't have to drill in too much into various constructors to see the type.

You might just want to write a helper function to do the Yes/No so you don't have to add a case for it.

```haskell
checkResult :: ExpectationCheckRequestValue -> FossaApiClientF a  -> FossaApiClientF a -> ApiResult a -> Maybe (ApiResult a)
checkResult Yes a b resp = resp <$ guard (a == b)
checkResult No _ _ resp = pure resp
```